docs/DocGenerator/README.md

```diff
@@ -0,0 +1,205 @@
+# Wazuh QA DocGenerator
+Wazuh - Quality assurance automation self-contained documentation parsing tool
+
+## Rational
+Wazuh QA documentation is designed to be self-contained into the source code of each test.
+
+DocGenerator is the tool in charge of parsing the documentation block from each source code and generate data capable
+to be indexed and displayed.
+
+## Design
+### Input
+DocGenerator parses the information from a QA repository with test files containing a specific comment format:
+
+Each test file has self-contained documentation comment blocks in **YAML** format.
+
+These blocks can contain **Mandatory**, **Optional**, and **Ignored** fields.
+In the [Configuration section](#configuration) it's explained how to configure these fields.
+- **Mandatory** fields must be present in the documentation block and will be dumped into the final
+documentation output.
+- **Optional** fields, if present, will be parsed and dumped into the final documentation output.
+- **Ignored** fields are the ones not listed in the configuration file. They can be added by the developer to explain
+a functionality of the module or a method without transferring this information to the final documentation output.
```

Comment:
I would like something like: "the functionality" and "without including this information into the output file"

Reply:
Done